### PR TITLE
[ISSUE #148] support authentication capability for polaris

### LINKED ISSUE #148: Improve the data operation authority control of the polaris console
**What is the feature you want to add?**

Improve the data operation authority control of the polaris console

**Why do you want to add this feature?**

At present, anyone can operate the console. In fact, there will be a relatively large security problem. Therefore, it is necessary to enhance the control of the polaris console for data operation permissions, add a polaris built-in permission implementation model, or implement an LDAP protocol Docking the auth plugin inside polaris

**How to implement this feature?**

**Additional context**

### PR DESCRIPTION
**Pull request by chuntaojun**
**Please provide issue(s) of this PR:**
Fixes #148 

- Implemented user management
- Implemented user group management
- Implemented authentication policy management

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration
- [ ] Docs
- [ ] Installation
- [ ] Performance and Scalability
- [x] Naming
- [ ] HealthCheck
- [ ] Test and Release
- [x] Auth

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: auth/defaultauth/user.go [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+
+	return svr.createUser(ctx, req)
+}
+
+func (svr *server) createUser(ctx context.Context, req *api.User) *api.Response {
+	requestID := utils.ParseRequestID(ctx)
+	platformID := utils.ParsePlatformID(ctx)
+
+	role := utils.ParseUserRole(ctx)
+	data, err := createUserModel(req, role)
```
Comment (houseme) [inline auth/defaultauth/user.go:109]:
```go
var(
	requestID = utils.ParseRequestID(ctx)
	platformID = utils.ParsePlatformID(ctx)

	role = utils.ParseUserRole(ctx)
	data, err = createUserModel(req, role)
)
```

File: cache/namespace.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"github.com/polarismesh/polaris-server/common/log"
+	"github.com/polarismesh/polaris-server/common/model"
+	"github.com/polarismesh/polaris-server/store"
+	"go.uber.org/zap"
```
Comment (houseme) [inline cache/namespace.go:28]: go fmt

File: auth/defaultauth/user.go [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+	"context"
+	"time"
+
+	api "github.com/polarismesh/polaris-server/common/api/v1"
+	"github.com/polarismesh/polaris-server/common/log"
+	"github.com/polarismesh/polaris-server/common/model"
+	commontime "github.com/polarismesh/polaris-server/common/time"
+	"github.com/polarismesh/polaris-server/common/utils"
+	"github.com/polarismesh/polaris-server/plugin"
+	"go.uber.org/zap"
```
[the remark this answers is not in the document]
Comment (chuntaojun) [reply auth/defaultauth/user.go:30]: 已经执行过 import 的格式化逻辑了，这里 polaris-server的格式化逻辑还没固定下来，以后看看怎么搞

**Review by andrewshan — APPROVED** [review]
OK

File: apiserver/eurekaserver/access.go [hunk 4]
```diff
@@ -320,9 +322,22 @@ func (h *EurekaServer) RegisterApplication(req *restful.Request, rsp *restful.Re
 	if !checkRegisterRequest(registrationRequest, req, rsp) {
 		return
 	}
+
+	// r.Header.Set("Authorization", "Basic "+basicAuth(username, password))
```
Comment (andrewshan) [inline apiserver/eurekaserver/access.go:326]: 这个头信息为啥要注释掉？

File: polaris-server.yaml [hunk 5]
```diff
@@ -227,26 +235,29 @@ cache:
     - name: routingConfig # 加载路由数据
     - name: rateLimitConfig # 加载限流数据
     - name: circuitBreakerConfig # 加载熔断数据
+    - name: users
```
Comment (andrewshan) [inline polaris-server.yaml:238]: 这里把注解加上

File: service/api.go [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+ 
+	 "github.com/polarismesh/polaris-server/cache"
+	 "github.com/polarismesh/polaris-server/common/api/l5"
+	 api "github.com/polarismesh/polaris-server/common/api/v1"
+	 "github.com/polarismesh/polaris-server/common/model"
+ )
+ 
+ // CircuitBreakerServer
+ type CircuitBreakerOperateServer interface {
+	 CreateCircuitBreakers(ctx context.Context, req []*api.CircuitBreaker) *api.BatchWriteResponse
```
Comment (andrewshan) [inline service/api.go:31]: 方法的注释要加上

**Review by andrewshan — APPROVED** [review]
OK

**Pull request merged**